### ISSUE TITLE
Key value of $group operator must be a string (unicode does not work).

### ISSUE THREAD

**Issue by polosatyi** [mongomock/mongomock#221]
If I use code (Python 2.7):

```

from __future__ import unicode_literals
aggregation = [{'$group': {'_id': {'cid': '$cid'}}}]
aggregated_data = db.events.aggregate(aggregation)
```

then I get an error:

_File "/usr/local/lib/python2.7/dist-packages/mongomock/collection.py", line 1387, in aggregate
    out_collection = sorted(out_collection, key=itemgetter(*group_func_keys))
TypeError: itemgetter expected 1 arguments, got 0_

I have to do here so:

```

from __future__ import unicode_literals
aggregation = [{'$group': {'_id': {'cid': b'$cid'}}}]
aggregated_data = db.events.aggregate(aggregation)
```

**Comment by polosatyi** [comment]
https://github.com/vmalloc/mongomock/pull/222

**Comment by pcorpet** [comment]
Closed by #236
